### PR TITLE
fix: enum-case mistaken for const invocation

### PR DESCRIPTION
**Pull request by SpacePossum**
closes https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/8189

### CODE REVIEW

File: src/Tokenizer/TokensAnalyzer.php [hunk 1]
```diff
@@ -375,10 +375,26 @@ public function isConstantInvocation(int $index): bool
...
         }
 
+        if ($this->tokens[$prevIndex]->isGivenKind([T_AS, T_CONST, T_DOUBLE_COLON, T_FUNCTION, T_GOTO, CT::T_GROUP_IMPORT_BRACE_OPEN, CT::T_TYPE_COLON, CT::T_TYPE_ALTERNATION, CT::T_TYPE_INTERSECTION]) || $this->tokens[$prevIndex]->isObjectOperator()) {
+            return false;
+        }
+
+        if (
+            $this->tokens[$prevIndex]->isGivenKind(T_CASE)
+            && \defined('T_ENUM')
+            && $this->tokens->isAllTokenKindsFound([T_ENUM])
```
Comment (SpacePossum) [inline src/Tokenizer/TokensAnalyzer.php:389]: I think adding a dedicated token name for the `T_CASE` of `T_ENUM`'s (something like `CT::ENUM_CASE`) in the next major of the tool/tokenizer would be good for performance and simplify the code, but not for now.

Comment (keradus) [reply src/Tokenizer/TokensAnalyzer.php:389]: can you send a draft for it, pls ?

Comment (SpacePossum) [reply src/Tokenizer/TokensAnalyzer.php:389]: While an interesting topic I regretfully cannot commit to it as I'm not sure I'll find the time anytime soon. Note that replacing the token-type with a new the CT one is a BC break so maybe better be worked on when v4 get in scope.

Comment (keradus) [reply src/Tokenizer/TokensAnalyzer.php:389]: ack.
i suspect it will easily get forgotten

**Review by kubawerlos — APPROVED** [review]

**Comment by kubawerlos** [comment]
Thanks @SpacePossum

File: src/Tokenizer/TokensAnalyzer.php (hunk 1 above)
Comment (Wirone) [reply src/Tokenizer/TokensAnalyzer.php:389]: @keradus I've created #8201 so we won't forget. I've added it to "long-term-ideas" milestone to not be handled by Stale Bot, but probably we need dedicated milestone for v4.

**Pull request merged**